src/Mapping/Driver/DatabaseDriver.php

```diff
@@ -274,15 +276,24 @@ private function reverseEngineerMappingFromDatabase(): void
                 $allForeignKeyColumns = array_merge($allForeignKeyColumns, self::getReferencingColumnNames($foreignKey));
             }
 
-            $primaryKey = $table->getPrimaryKey();
+            if (class_exists(PrimaryKeyConstraint::class)) {
```

Comment:
Would it make more sense to check if the method exists here?
```suggestion
            if (method_exists($table, 'getPrimaryKeyConstraint')) {
```
Not only the code will be easier to understand, it will likely spare some static analysis errors.